### ISSUE TITLE
Get Authoritative AZ

### ISSUE THREAD

**Issue by colinbjohnson** [colinbjohnson/aws-missing-tools#53]
AWS seemingly randomly assigns AZs - one customer's us-east-1a might be another customer's us-east-1b - a tool to determine the actual AZ would be useful as some AZs allows particular instances types or offer potentially improved performance.

**Comment by Krylon360** [comment]
getting the AZ from the metadata is the best viable option. the response from curl is the correct AZ

**Comment by whereisaaron** [comment]
I don't think that is correct. It would create chaos for outage notifications for the start. An AZ '1a' is the same data center for everybody.

What is true is that the initial default subnet for a VPC are assigned randomly to AZs, so e.g. one person's 172.31.0.0/20 will be '1a' where as someone else's 172.31.0.0/20 will be '1b'. This is true is all regions, not just 'us-east-1'.

You can delete and recreate the subnets if you want to make them all line up, but using the AZ metadata is a better way to go.

Interestingly the subsequent subnets are assigned in order. So e.g. is 172.31.0.0/20 is '1b' then 172.31.16.0/20 will be '1c' and 172.31.32.0/20 is '1e' and 172.31.48.0/20 is '1a'. But again, no need to rely on that, just is the metadata (Luke).
